integration-tests/oidc-wiremock-providers/src/main/java/io/quarkus/it/oidc/providers/ProvidersResource.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkus.it.oidc.providers;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.SecurityContext;
+
+import io.quarkus.oidc.AuthorizationCodeFlow;
+import io.quarkus.oidc.UserInfo;
+
+@Path("providers")
+public class ProvidersResource {
+
+    public record ProvidersResponseDto(String userPrincipalName, String userInfoEmail) {
+    }
+
+    @Inject
+    UserInfo userInfo;
```

Comment:
Hi Michal, @michalvavrik,  you should have IdToken injected here, we don't need `UserInfo` for Slack, I also propose to have `SlackResource`  at the root `/slack` path, just to avoid mixing in various provider expectations here going forward...
Sorry, hope I'm not too annoying :-)

Reply:
> Hi Michal, @michalvavrik, you should have IdToken injected here, we don't need UserInfo for Slack out of the box, I also propose to have SlackResource at the root /slack path, just to avoid mixing in various provider expectations here going forward...

OK.

> Sorry, hope I'm not too annoying :-)

I don't really mind, I am fine with making changes for weeks (not that this is case, just explanation how much I am OK with this). That's just part of the business.